app/src/main/java/com/quipper/android/apollofrontpage/repository/impl/PostsRepository.kt

```diff
@@ -0,0 +1,24 @@
+package com.quipper.android.apollofrontpage.repository.impl
+
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.rx2.rxMutate
+import com.apollographql.apollo.rx2.rxQuery
+import com.quipper.android.apollofrontpage.AllPostsQuery
+import com.quipper.android.apollofrontpage.UpvotePostMutation
+import com.quipper.android.apollofrontpage.repository.PostsRepository
+import io.reactivex.Observable
+import io.reactivex.Single
+
+class PostsRepository(
+    private val apolloClient: ApolloClient
+) : PostsRepository {
+
+    override fun getPosts(): Observable<Response<AllPostsQuery.Data>> {
```

Comment:
[NITS]
We should use `Single` instead of `Observable`. Because remote API commit once.

Reply:
[NITS]
```suggestion
    override fun getPosts(): Observable<AllPostsQuery.Data> {
```
We'd better remove `Reponse` ?

Reply:
Thanks! Sorry seems I forgot that. Addressed.